lightning/src/ln/functional_test_utils.rs

```diff
@@ -2363,70 +2351,22 @@ macro_rules! expect_htlc_handling_failed_destinations {
 	}};
 }
 
-/// Checks that an [`Event::PendingHTLCsForwardable`] is available in the given events and, if
-/// there are any [`Event::HTLCHandlingFailed`] events their [`HTLCHandlingFailureType`] is included in the
-/// `expected_failures` set.
-pub fn expect_pending_htlcs_forwardable_conditions(
+/// Checks that, if there are any [`Event::HTLCHandlingFailed`] events, their
+/// [`HTLCHandlingFailureType`] is included in the `expected_failures` set.
+pub fn expect_htlc_failure_conditions(
 	events: Vec<Event>, expected_failures: &[HTLCHandlingFailureType],
 ) {
-	let count = expected_failures.len() + 1;
-	assert_eq!(events.len(), count);
-	assert!(events
-		.iter()
-		.find(|event| matches!(event, Event::PendingHTLCsForwardable { .. }))
-		.is_some());
+	assert_eq!(events.len(), expected_failures.len());
 	if expected_failures.len() > 0 {
 		expect_htlc_handling_failed_destinations!(events, expected_failures)
 	}
 }
 
 #[macro_export]
-/// Clears (and ignores) a PendingHTLCsForwardable event
-///
-/// Don't use this, call [`expect_pending_htlcs_forwardable_conditions()`] with an empty failure
-/// set instead.
-macro_rules! expect_pending_htlcs_forwardable_ignore {
-	($node: expr) => {
-		$crate::ln::functional_test_utils::expect_pending_htlcs_forwardable_conditions(
-			$node.node.get_and_clear_pending_events(),
-			&[],
-		);
-	};
-}
-
-#[macro_export]
-/// Clears (and ignores) PendingHTLCsForwardable and HTLCHandlingFailed events
-///
-/// Don't use this, call [`expect_pending_htlcs_forwardable_conditions()`] instead.
-macro_rules! expect_pending_htlcs_forwardable_and_htlc_handling_failed_ignore {
-	($node: expr, $expected_failures: expr) => {
-		$crate::ln::functional_test_utils::expect_pending_htlcs_forwardable_conditions(
-			$node.node.get_and_clear_pending_events(),
-			&$expected_failures,
-		);
-	};
-}
-
-#[macro_export]
-/// Handles a PendingHTLCsForwardable event
-macro_rules! expect_pending_htlcs_forwardable {
-	($node: expr) => {{
-		$crate::ln::functional_test_utils::expect_pending_htlcs_forwardable_conditions(
-			$node.node.get_and_clear_pending_events(),
-			&[],
-		);
-		$node.node.process_pending_htlc_forwards();
-
-		// Ensure process_pending_htlc_forwards is idempotent.
-		$node.node.process_pending_htlc_forwards();
-	}};
-}
-
-#[macro_export]
-/// Handles a PendingHTLCsForwardable and HTLCHandlingFailed event
-macro_rules! expect_pending_htlcs_forwardable_and_htlc_handling_failed {
+/// Processes any HTLC forwards and handles an expected [`Event::HTLCHandlingFailed`].
+macro_rules! process_htlcs_and_expect_htlc_handling_failed {
```

Comment:
Now that we have the notifier, can we revert a bunch of the diff in tests and change `expect_pending_htlcs_forwardable` to call `ChannelManager::is_pending_htlc_processing`? Seems like strictly better test coverage and easier review. Matt got me overthinking the test coverage changes...

Reply:
Darn, I thought it might work to just drop the fixup that is specific to the removed macro.

Reply:
I am in favor if it reduces the diff and increases coverage.

Does this need to be a macro btw?